pinot-core/src/main/java/org/apache/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -132,47 +129,40 @@ public FixedByteSingleColumnMultiValueReaderWriter(int maxNumberOfMultiValuesPer
     _maxNumberOfMultiValuesPerRow = maxNumberOfMultiValuesPerRow;
     _headerSize = rowCountPerChunk * SIZE_OF_INT * NUM_COLS_IN_HEADER;
     _rowCountPerChunk = rowCountPerChunk;
-    addHeaderBuffers();
+    addHeaderBuffer();
     //at least create space for million entries, which for INT translates into 4mb buffer
     _incrementalCapacity = incrementalCapacity;
-    addDataBuffers(initialCapacity);
+    addDataBuffer(initialCapacity);
     //init(_rowCountPerChunk, _columnSizeInBytes, _maxNumberOfMultiValuesPerRow, initialCapacity, _incrementalCapacity);
   }
 
-  private void addHeaderBuffers() {
+  private void addHeaderBuffer() {
     LOGGER.info("Allocating header buffer of size {} for: {}", _headerSize, _context);
-    _headerBuffer = _memoryManager.allocate(_headerSize, _context);
+    // NOTE: PinotDataBuffer is tracked in the PinotDataBufferMemoryManager. No need to track it inside the class.
+    PinotDataBuffer headerBuffer = _memoryManager.allocate(_headerSize, _context);
```

Comment:
Not sure I understand this. is headerBuffer(s) alllocated here closed anywhere? Same q with databBuffers . The reader and writer classes we construct here do not close these buffers, right?

Reply:
MemoryManager allocates and tracks all the buffers. It releases them when the segment is closed.